azure-pipelines.yml

```diff
@@ -21,6 +13,25 @@ jobs:
   steps:
   - template: azure-pipelines-steps.yml
 
+  - script: npm pack
+    displayName: 'npm pack'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+
+  - task: CopyFiles@2
+    inputs:
+      sourceFolder: '$(Build.SourcesDirectory)'
+      contents: '*.tgz' 
+      targetFolder: $(Build.ArtifactStagingDirectory)
+    displayName: 'Copy npm package to staging'
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+
+  - task: PublishBuildArtifacts@1
```

Comment:
Thoughts on, instead of doing this, just having a seperate pipeline that's manually triggered that publishes directly to npm (and maybe creates a GitHub release too)?

Reply:
You mean duplicate all of the steps before again?

Reply:
Yeah. Its very inexpensive to build these libraries, and its not a bad thing to have an extra validation before publishing

Reply:
I mostly just think it would simplify things though